crates/matrix-sdk/src/widget_api/error.rs

```diff
@@ -21,3 +23,12 @@ pub enum Error {
     #[error("Failed to perform an operation")]
     Other,
 }
+impl Error {
+    pub fn to_description_string(&self) -> String {
+        match self {
+            Error::InvalidJSON(e) => format!("{}: {}",self.to_string(), e),
+            Error::WidgetError(e) => format!("{}: {}",self.to_string(), e),
+            _ => self.to_string(),
+        }
+    }
+}
```

Comment:
Is it necessary? `thiserror` provides a decent implementation for the types and we could define our formatting string inside the `#[error(...)]`.